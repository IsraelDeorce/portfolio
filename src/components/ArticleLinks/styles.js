import styled from 'styled-components'

export const ArticleLinks = styled.ul`
  display: flex;
  flex-direction: column;
  padding-left: 1em;
  & > * {
    color: ${({ theme }) => theme.c_brackets};
    &:hover {
      color: ${({ theme }) => theme.c_varType};
    }
    padding-bottom: 1em;
  }
`
