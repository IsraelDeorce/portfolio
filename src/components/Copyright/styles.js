import styled from 'styled-components'

export const Copyright = styled.div`
  display: flex;
  justify-content: center;
`
