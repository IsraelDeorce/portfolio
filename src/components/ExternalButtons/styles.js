import styled from 'styled-components'

export const ExternalButtons = styled.div`
  display: flex;
  flex-direction: column;
  align-items: center;
  margin-top: 20px;
  font-size: 1.25rem;
`
