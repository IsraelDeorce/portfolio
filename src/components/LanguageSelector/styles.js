import styled from 'styled-components'

export const LanguageSelector = styled.div`
  display: flex;
  align-items: center;
  cursor: pointer;
  color: ${({ theme }) => theme.c_brackets};
`

export const LanguageCode = styled.div`
  display: flex;
  flex-direction: column;
  align-items: center;
`