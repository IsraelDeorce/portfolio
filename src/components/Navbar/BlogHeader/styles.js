import styled from 'styled-components'

export const BlogHeader = styled.nav`
  position: fixed;
  top: 0;
  left: 0;
  right: 0;
  z-index: 1;
  display: flex;
  align-items: center;
  font-weight: 700;
  background-color: ${({ theme }) => theme.c_menu};
  color: ${({ theme }) => theme.c_brackets};
  padding: 0 1.5em;
  min-height: 60px;
  & > *:not(:last-child) {
    margin-right: 0.75em;
  }
  @media (min-width: 992px) {
    display: none;
  }
`
export const Hamburguer = styled.span`
  display: flex;
  align-items: center;
  cursor: pointer;
  flex-shrink: 0;
  &:hover {
    color: ${({ theme }) => theme.c_varType};
  }
`
