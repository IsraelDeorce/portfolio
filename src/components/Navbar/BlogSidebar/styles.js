import styled from 'styled-components'

export const Background = styled.div`
  opacity: 0.7;
  background-color: ${({ theme }) => theme.c_background};
  position:fixed;
  width:100%;
  height:100%;
  top:0px;
  left:0px;
  @media (min-width: 992px) {
    display: none;
  }
`

export const BlogSidebar = styled.div`
  position: fixed;
  top: 0;
  left: 0;
  right: 0;
  display: flex;
  flex-direction: column;
  background-color: ${({ theme }) => theme.c_menu};
  height: 100vh;
  width: 65vw;
  z-index: 2;
  box-shadow: 0 5px 30px ${({ theme }) => theme.c_tab};
  padding: 0 1em 1em 1em;
  & > * {
    width: 100%;
  }
  & > :last-child {
    align-self: center;
  }
  @media (min-width: 992px) {
    position: static;
    top: auto;
    left: auto;
    right: auto;
    height: auto;
    width: 285px;
    min-width: 285px;
    box-shadow: none;
  }
`

export const Header = styled.div`
  display: flex;
  flex-wrap: wrap;
  align-items: center;
  min-height: 60px;
  border-bottom: 2px solid gray;
  & > *:not(:last-child) {
    padding: 0.35em;
  }
`

export const Preferences = styled.div`
  display: flex;
  align-items: center;
  & > *:not(:last-child) {
    margin-right: 0.75em;
  }
`
