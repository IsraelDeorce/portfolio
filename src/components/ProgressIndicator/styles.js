import styled from 'styled-components'

export const ProgressIndicator = styled.div`
  display: flex;
  align-items: center;
`

export const ProgressBar = styled.div`
  display: flex;
  border-radius: 25px;
  border: 4px solid ${({ theme }) => theme.c_function};
  min-width: 180px;
  height: 30px;
  padding: 3.5px;
  margin-right: 0.5em;
  @media (min-width: 576px) {
    min-width: 350px;
  }
`

export const Progress = styled.div`
  background-color: ${({ theme }) => theme.c_varDeclaration};
  border-radius: 25px;
  min-width: ${({ percentage }) => percentage}%;
`
