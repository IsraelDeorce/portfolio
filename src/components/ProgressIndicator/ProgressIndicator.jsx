import * as S from './styles'

function ProgressIndicator({ percentage = 0, description = '' }) {

  function handlePercentage() {
    if (percentage < 10) return 10
    if (percentage > 100) return 100
    return percentage
  }

  return (
    <S.ProgressIndicator>
      <S.ProgressBar>
        <S.Progress percentage={handlePercentage()} />
      </S.ProgressBar>
      {description && <div>{description}</div>}
    </S.ProgressIndicator>
  )
}

export default ProgressIndicator
