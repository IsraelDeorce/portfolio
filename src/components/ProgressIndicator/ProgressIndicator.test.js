import ProgressIndicator from './ProgressIndicator'
import * as S from './styles'

describe('<ProgressIndicator/>', () => {

  it.each([
    [-1], [0], [9], [10]
  ])('renders a progress bar with progress as 10 when percentage is %s', (percentage) => {
    expect(shallow(<ProgressIndicator percentage={percentage} />)
    .find(S.Progress).prop('percentage')).toBe(10)
  })

  it.each([
    [100], [101]
  ])('renders a progress bar with progress as 100 when percentage is %s', (percentage) => {
    expect(shallow(<ProgressIndicator percentage={percentage} />)
    .find(S.Progress).prop('percentage')).toBe(100)
  })

  it.each([
    [11], [99]
  ])('renders a progress bar with progress as %s when percentage is %s', (percentage) => {
    expect(shallow(<ProgressIndicator percentage={percentage} />)
    .find(S.Progress).prop('percentage')).toBe(percentage)
  })

  describe('Snapshot', () => {
    it('renders without crashing using shallow', () => {
      expect(shallow(<ProgressIndicator />)).toMatchSnapshot()
    })
  })
})
