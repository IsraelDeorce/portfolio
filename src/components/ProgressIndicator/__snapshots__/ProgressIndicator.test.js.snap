// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ProgressIndicator/> Snapshot renders without crashing using mount 1`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.c1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  border-radius: 25px;
  border: 4px solid;
  min-width: 180px;
  height: 30px;
  padding: 3.5px;
  margin-right: 0.5em;
}

.c2 {
  border-radius: 25px;
  min-width: 10%;
}

@media (min-width:576px) {
  .c1 {
    min-width: 350px;
  }
}

<ProgressIndicator>
  <styled.div>
    <div
      className="c0"
    >
      <styled.div>
        <div
          className="c1"
        >
          <styled.div
            percentage={10}
          >
            <div
              className="c2"
            />
          </styled.div>
        </div>
      </styled.div>
    </div>
  </styled.div>
</ProgressIndicator>
`;

exports[`<ProgressIndicator/> Snapshot renders without crashing using render 1`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.c1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  border-radius: 25px;
  border: 4px solid;
  min-width: 180px;
  height: 30px;
  padding: 3.5px;
  margin-right: 0.5em;
}

.c2 {
  border-radius: 25px;
  min-width: 10%;
}

@media (min-width:576px) {
  .c1 {
    min-width: 350px;
  }
}

<div
  class="c0"
>
  <div
    class="c1"
  >
    <div
      class="c2"
    />
  </div>
</div>
`;

exports[`<ProgressIndicator/> Snapshot renders without crashing using shallow 1`] = `
<styled.div>
  <styled.div>
    <styled.div
      percentage={10}
    />
  </styled.div>
</styled.div>
`;
